plumbing/format/packfile/delta_selector.go

```diff
@@ -138,16 +139,33 @@ func (dw *deltaSelector) fixAndBreakChains(objectsToPack []*ObjectToPack) error
 	for _, otp := range objectsToPack {
 		m[otp.Hash()] = otp
 	}
+	numJobs := len(objectsToPack)
+	jobs := make(chan *ObjectToPack, numJobs)
+	results := make(chan error, numJobs)
 
+	for w := 1; w <= workers; w++ {
+		go dw.fixAndBreakChainsWorker(w, m, jobs, results)
+	}
 	for _, otp := range objectsToPack {
-		if err := dw.fixAndBreakChainsOne(m, otp); err != nil {
-			return err
-		}
+		jobs <- otp
 	}
 
+	close(jobs)
+
 	return nil
 }
 
+func (dw *deltaSelector) fixAndBreakChainsWorker(id int, objectsToPack map[plumbing.Hash]*ObjectToPack, jobs <-chan *ObjectToPack, results chan<- error) {
+	for j := range jobs {
+		err := dw.fixAndBreakChainsOne(objectsToPack, j)
+		if err != nil {
+			results <- err
+		} else {
+			results <- nil
+		}
```

Comment:
You should be able to just do `results <- err`, or `results <- dw.fixAndBreakChainsOne(objectsToPack, j) `